pymeasure/instruments/hp/hp3478A.py

```diff
@@ -552,13 +412,12 @@ def SRQ_mask(self):
         =========  ==========================
 
         """
-        mask = self.decode_status(self.get_status(), "SRQ")
-        return mask
+        mask = self.status.SRQ
+        return self.SRQ(mask)
```

Comment:
I think, you can have the same behavior using a nested `ctypes` bitfield class, but I haven't tried it.

Reply:
> I think, you can have the same behavior using a nested ctypes bitfield class, but I haven't tried it.
I have not tried it either, and I was not able to find a good example of nested ctyoes bitfields (yet)

Let me see if I can find some better way to simplify this.

Reply:
```
import ctypes
c_uint8 = ctypes.c_uint8

class F(ctypes.BigEndianStructure):
	_fields_ = [
        ("SRQ1", c_uint8, 1),
        ("SRQ2", c_uint8, 1),
     ]
class C(ctypes.BigEndianStructure):
	_fields_ = [
        ("byte1", c_uint8, 8),
        ("SRQ", F),
     ]
a=C.from_buffer(bytearray([0xff, 0x40]))
print(a.byte1)
print(a.SRQ.SRQ2)
print(a.SRQ) # This is not accessed as number
```
I have made this example, please note that SRQ field cannot be accessed as byte (unless a union is used)

Reply:
Thank you for the example, I'll have a look at it during the week and hopefully have an update torards the end of the week

Reply:
I think I found a good compromise,